### PR TITLE
extend bucket_internal to SAMPLE generation mode

### PR DESCRIPTION
**Pull request by xt574chen**
# What does this PR do?

REF: https://github.com/HabanaAI/optimum-habana-fork/pull/84

Extend function #24 to sample mode.
image

The command to reproduce performance is as follows:
`python ../gaudi_spawn.py --use_deepspeed --world_size 4 run_generation.py --model_name_or_path meta-llama/Llama-2-70b-hf --use_hpu_graphs --use_kv_cache --max_input_tokens 128 --max_new_tokens 2048 --batch_size 240 --attn_softmax_bf16  --trim_logits --bf16 --reuse_cache --warmup 1 --n_iterations 1 --limit_hpu_graphs --do_sample --bucket_size 256 --bucket_internal`

### CODE REVIEW

**Comment by puneeshkhanna** [comment]
Changes look good to me. Same as we have in greedy search and enables bucketing in sampling mode too.
We can merge this.

**Comment by xt574chen** [comment]
@regisss Could u help review and merge it?

**Review by ssarkar2 — APPROVED** [review]

**Review by regisss — APPROVED** [review]
